nameko/testing/waiting.py

```diff
@@ -0,0 +1,63 @@
+""" https://github.com/mattbennett/call-waiting
+"""
+import sys
+from contextlib import contextmanager
+from threading import Semaphore
+
+import six
+from mock import patch
+
+
+class WaitResult(object):
+    res = None
```

Comment:
A sentinel? Completely agree.

Raising an exception seems like a nice idea, because otherwise you force the user to compare the value to the sentinel (or conditionally check it's ready before calling `get`)